staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -378,9 +378,10 @@ func structFromUnstructured(sv, dv reflect.Value) error {
 	for i := 0; i < dt.NumField(); i++ {
 		fieldInfo := fieldInfoFromField(dt, i)
 		fv := dv.Field(i)
-
-		if len(fieldInfo.name) == 0 {
-			// This field is inlined.
+		// This field is inlined.
+		// if not found fieldInfo.name of Anonymous field in json,the json is inline mode for this Anonymous field
+		_, ok := sv.Interface().(map[string]interface{})[fieldInfo.name]
```

Comment:
I would have expected this to trigger off of the `json` tag in the struct, not off the presence/absence of the field name in the JSON data

Reply:
Yes. But when there are no json information for StructB, the SturctB will not convert success. Beacuse len(fieldInfo.name) == 0 will not be satisfied (The  fieldInfo.name is string structB).